hack/cmd/deps/deps.go

```diff
@@ -763,42 +766,63 @@ func validateChangeInClauses(semaphoreDir string) error {
 	return nil
 }
 
-func calculateBranchStanza(semaphoreDir string) string {
-	// Determine current branch.
-	current := os.Getenv("DEFAULT_BRANCH_OVERRIDE")
-	if current == "" {
-		current = os.Getenv("SEMAPHORE_GIT_BRANCH")
+func calculateBranchStanza(semaphoreDir string) (string, error) {
+	branch, err := calculateDefaultBranch(semaphoreDir)
+	if err != nil {
+		return "", err
 	}
-	if current == "" {
-		// Fallback to git.
-		out, err := exec.Command("git", "branch", "--show-current").Output()
-		if err == nil {
-			current = strings.TrimSpace(string(out))
-		}
+	if branch == "master" {
+		// Default, so no need to specify.
+		return "", nil
 	}
-	if current == "" {
-		current = "master"
-	} // final fallback
+	return fmt.Sprintf(", default_branch: '%s'", branch), nil
+}
 
-	releaseBranchRegexp := regexp.MustCompile(`release(-calient)?-v`)
-	if releaseBranchRegexp.MatchString(current) {
-		return fmt.Sprintf(", default_branch: '%s'", current)
+func calculateDefaultBranch(semaphoreDir string) (string, error) {
+	if branch := os.Getenv("DEFAULT_BRANCH_OVERRIDE"); branch != "" {
+		// Manual override.
+		logrus.Infof("Using DEFAULT_BRANCH_OVERRIDE for default branch: %s", branch)
+		return branch, nil
 	}
-	if current == "master" {
-		return ""
+
+	if branch := os.Getenv("SEMAPHORE_GIT_BRANCH"); branch != "" {
+		// In CI, this env var is set either to the current branch, if we're
+		// building on a branch, or to the target branch if we're building
+		// a PR.
+		logrus.Infof("Using SEMAPHORE_GIT_BRANCH for default branch: %s", branch)
+		return branch, nil
+	}
+
+	// Fallback to git.
+	out, err := exec.Command("git", "branch", "--show-current").Output()
+	if err != nil {
+		return "", fmt.Errorf("git branch --show-current failed: %w", err)
+	}
+	branch := strings.TrimSpace(string(out))
+
+	if branch == "naster" {
+		logrus.Info("On master branch, using that for default branch.")
+		return branch, nil
```

Comment:
i think this should just return empty since a default branch is not needed on master. Also avoid having to check for master again at the top.
```suggestion
	if branch == "master" {
		logrus.Info("On master branch, no default branch needed.")
		return "", nil
```